src/Illuminate/Validation/ValidationRuleParser.php

```diff
@@ -79,26 +79,34 @@ protected function explodeRules($rules)
      * Explode the explicit rule into an array if necessary.
      *
      * @param  mixed  $rule
+     * @param  string  $attribute
      * @return array
      */
-    protected function explodeExplicitRule($rule)
+    protected function explodeExplicitRule($rule, $attribute)
     {
         if (is_string($rule)) {
             return explode('|', $rule);
         } elseif (is_object($rule)) {
-            return [$this->prepareRule($rule)];
+            return Arr::wrap($this->prepareRule($rule, $attribute));
         }
 
-        return array_map([$this, 'prepareRule'], $rule);
+        $attributes = array_fill(
+            array_key_first($rule), count($rule), $attribute
+        );
+
+        return array_map(
+            [$this, 'prepareRule'], $rule, $attributes
+        );
     }
```

Comment:
Can you explain the `array_fill` and `array_map` above and explain why they are needed?

Reply:
Yes absolutely.

I use `array_fill` here to generate an array filled with the exact same attribute, which will be equal to the number of rules that are being iterated over. This allows `array_map` to pass the attribute into the second parameter of the `parepareRule` method.

This is mainly for convenience because `array_map` doesn't support passing secondary arguments into the map method.